tests/realtikvtest/brietest/scheduler_test.go

```diff
@@ -0,0 +1,438 @@
+			continue
+		}
+
+		t.Logf("Attempt %d: Got %d scheduler rules", i+1, len(rules))
+
+		// Check if any rule has more key ranges than baseline
+		hasNewRanges := false
+
+		for _, rule := range rules {
+			ruleKey := fmt.Sprintf("%s/%s", rule.GroupID, rule.ID)
+			keyRanges, err := extractKeyRangesFromRule(rule)
+			if err == nil {
+				// Compare with baseline
+				if baselineRanges, exists := baselineKeyRanges[ruleKey]; exists {
+					if len(keyRanges) > len(baselineRanges) {
+						hasNewRanges = true
+						t.Logf("Detected new key ranges in rule %s! Baseline: %d, Current: %d",
+							ruleKey, len(baselineRanges), len(keyRanges))
+					}
+				}
+			}
+		}
+
+		if hasNewRanges {
+			finalRules = rules
+			t.Logf("Scheduler pausing detected - new key ranges added!")
+			break
+		}
+
+		if i == maxRetries-1 {
+			finalRules = rules
+			t.Logf("Reached max retries. Using final rule set.")
+		}
+	}
+
+	return finalRules
+}
+
+// TestLogRestoreFineGrainedSchedulerPausing tests that log restore uses fine-grained
+// scheduler pausing when filters are specified
+func TestLogRestoreFineGrainedSchedulerPausing(t *testing.T) {
+	kit := NewLogBackupKit(t)
+	taskName := "test-fine-grained-scheduler"
+
+	setupTestData(kit, taskName)
+	cleanupTestData(kit)
+
+	// Get baseline scheduler rules
+	t.Log("Getting baseline scheduler rules...")
+	baselineRules, err := getPDSchedulerRules(t, "127.0.0.1:2379")
+	require.NoError(t, err)
+	baselineKeyRanges := analyzeSchedulerRules(t, baselineRules, "BASELINE SCHEDULER RULES (before restore)")
+
+	// Create channel to capture results from the callback
+	//rulesChan := make(chan []SchedulerRule, 1)
+
+	// Enable failpoint with callback that checks PD scheduler status
+	//require.NoError(t, failpoint.EnableCall("github.com/pingcap/tidb/br/pkg/task/log-restore-scheduler-paused", func() {
+	//	t.Log("Failpoint triggered - checking PD scheduler rules")
+	//	finalRules := checkSchedulerPausingBehavior(t, baselineKeyRanges)
+	//	rulesChan <- finalRules
+	//}))
+	//defer failpoint.Disable("github.com/pingcap/tidb/br/pkg/task/log-restore-scheduler-paused")
+
+	// Run filtered restore
+	t.Log("Starting filtered restore...")
+	kit.RunStreamRestore(func(rc *task.RestoreConfig) {
+		kit.SetFilter(&rc.Config, "test.snapshot_table1", "test.log_table1")
+	})
+
+	// Wait for callback results
+	//select {
+	//case restoreRules := <-rulesChan:
+	//	t.Log("Analyzing scheduler rules during filtered restore:")
+	//	restoreKeyRanges := analyzeSchedulerRules(t, restoreRules, "SCHEDULER RULES DURING FILTERED RESTORE")
+	//	hasChanges := compareKeyRanges(t, baselineKeyRanges, restoreKeyRanges)
+	//
+	//	if hasChanges {
+	//		t.Log("SUCCESS: Fine-grained scheduler pausing detected during filtered restore")
+	//	} else {
+	//		t.Log("WARNING: No scheduler changes detected during filtered restore")
+	//	}
+	//
+	//case <-time.After(20 * time.Second):
+	//	t.Logf("Timeout waiting for failpoint callback")
+	//}
```

Comment:
Would you remove those temporary codes or just uncomment them if they are necessrary?